### ISSUE TITLE
Train Tiny Yolo V2 with COCO 

### ISSUE THREAD

**Issue by LouiseAbela** [AlexeyAB/darknet#569]
Hi,

I tried a lot of things and can't find out how to do it. When training with COCO but only 7 classes using yolov2-tiny as config and weights, and change the classes to 7 and filters to 60, the error stays around 2 and doesn't go down. Does anyone have a clue why please?

Thanks very much!

**Comment by AlexeyAB** [comment]
Hi,

* What number of `filters=` did you use in penultimate conv layer?
* What number of `classes=` and `num=` did you use in region-layer?
* What mAP can you get? `darknet detector map`...

**Comment by LouiseAbela** [comment]
- Filters = 512, and in last one 60
- Classes = 7 and num = 5 
- And the mAP after 55000 epochs is 0.000726, or 0.07 %

And thanks for answering :).

**Comment by AlexeyAB** [comment]
1. Check, do you use in yolov2-tiny.cfg?
```
batch=64
subdivisions=2
```
And removed any other lines with batch and subdivision?

2. Do you try to train for the first 7 classes?
```
person
bicycle
car
motorbike
aeroplane
bus
train
```

* If all are correct, the try to use in cfg-file
```
learning_rate=0.001
max_batches = 120000
policy=steps
steps=-1,100,80000,100000
scales=.1,10,.1,.1
```

**Comment by LouiseAbela** [comment]
Hi, 

1. Yes I am using that configuration and have those values for the batches, and have the testing ones commented.

2. Not the first 7 but  I did of course create the new labels with the new indexes, and double checked with the old that the annotations are correct.

   ```
   person
   bicycle
   car
   motorbike
   bus
   cat
   dog
  
3. I will try that today and get back to you :)

Thanks again !

**Comment by RushNuts** [comment]
Hi @LouiseAbela 
1. which model you use for training tiny with COCO dataset
2. what are the results of the training now?

**Comment by LouiseAbela** [comment]
@AlexeyAB I will let you know sorry didn't had the time to get to it, just started the training now.

@RushNuts 

1. I'm using the Yolov2-tiny.cfg (of course modified for just 7 classes) - I attached my config file here
2. Running it now as didn't had time to get to it during this week, will let you know when it finished 

```
[net]
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
max_batches = 120000
policy=steps
steps=-1,100,80000,100000
scales=.1,10,.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=512
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=60
activation=linear

[region]
anchors =  0.57273, 0.677385, 1.87446, 2.06253, 3.33843, 5.47434, 7.88282, 3.52778, 9.77052, 9.16828
bias_match=1
classes=7
coords=4
num=5
softmax=1
jitter=.2
rescore=0

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1
```

**Comment by RushNuts** [comment]
@LouiseAbela okay, thanks for your answer. I'll wait you. One more thing, which pre-trained model you use? darknet19_448?

**Comment by LouiseAbela** [comment]
@AlexeyAB @RushNuts, still the maP stays really low 👎. I don't know what to change more with it.

No no I used - yolov2-tiny.weights and then convert them to .conv.13, but I also trained it without weights it doesn't help. I try to debug it a little and see what is happening, maybe it's just a stupid mistake.

**Comment by RushNuts** [comment]
@LouiseAbela I have a similar problem. AVG does not go below 1.5, of this the percentage of detection is very low and many false positives. My .cfg file is the same as yours, also used .conv.13 after conversion...uh
